### ISSUE TITLE
Shell flyout items are displayed multiple times - if async added by code

### ISSUE THREAD

**Issue by christophbrunner** [dotnet/maui#12233]
### Description

If you add Shell Flyout items by code, they will be displayed multiple times on Windows.
It seems that with every `Items.Add(x)` all already existing items are added / drawn again.
The behaviour is only present when `Items.Add(x)` is called in an async method.

**Windwos:**
![Maui-AddFlyoutIItem-Async_Windows-small](https://user-images.githubusercontent.com/1324374/208773072-1e0c9ec9-4388-46ca-abb5-054af75383a6.png)

**Works with Ios / Android:**
![Maui-AddFlyoutIItem-Async_Ios-small](https://user-images.githubusercontent.com/1324374/208773097-f8a5e938-8bf9-4d13-a2cc-ae776a03fe78.png)
![Maui-AddFlyoutIItem-Async_Android-small](https://user-images.githubusercontent.com/1324374/208773088-6c4a34cd-3160-4521-afee-ef8506c3449a.png)

### Steps to Reproduce

1. Change `Shell.FlyoutBehavior` to `Flyout`
2. Add the following code on App.Shell.xmal.cs:

```
protected override async void OnAppearing()
{
    base.OnAppearing();
    await AddFlyoutItems();
}

public async Task AddFlyoutItems()
{
    // Get  flyout items from Web
    await Task.Delay(500);

    for (int i = 0; i < 3; i++)
    {
        var flyoutItem = new FlyoutItem
        {
            Title = $"{i}-item",
            Route = $"{i}-item"
        };
        flyoutItem.Items.Add(new ShellContent
        {
            Content = new MainPage(),
            Route = $"{i}-item"
        });
        Items.Add(flyoutItem);
    }
}
```

### Link to public reproduction project repository

https://github.com/christophbrunner/MauiAddShellItem

### Version with bug

7.0 (current)

### Last version that worked well

6.0.424

### Affected platforms

Windows

### Affected platform versions

10.0.17763.0

### Did you find any workaround?

Add items in a synchronous method.

**Comment by ArturWyszomirski** [comment]
I have the same issue in my project where my AddFlyoutItem method is subscribed to event handler. In this case it doesn't matter whether it is synchronous method or not. Everything used to worked fine a couple months ago so the issue must be related to recent updates.

**Comment by ArturWyszomirski** [comment]
Yes, it's fixed:)

**Comment by christophbrunner** [comment]
Is fixed in 7.0.81
